corehq/apps/domain/forms.py

```diff
@@ -1623,15 +1624,32 @@ def save(self, commit=True):
 
                 cancel_future_subscriptions(self.domain, datetime.date.today(), self.creating_user)
                 if self.current_subscription is not None:
-                    self.current_subscription.change_plan(
-                        self.plan_version,
-                        web_user=self.creating_user,
-                        adjustment_method=SubscriptionAdjustmentMethod.USER,
-                        service_type=SubscriptionType.PRODUCT,
-                        pro_bono_status=ProBonoStatus.NO,
-                        do_not_invoice=False,
-                        no_invoice_reason='',
-                    )
+                    if self.is_downgrade() and self.current_subscription.is_below_minimum_subscription:
+                        self.current_subscription.update_subscription(
+                            date_start=self.current_subscription.date_start,
+                            date_end=self.current_subscription.date_start + datetime.timedelta(days=30)
+                        )
+                        Subscription.new_domain_subscription(
+                            account=self.account,
+                            domain=self.domain,
+                            plan_version=self.plan_version,
+                            date_start=self.current_subscription.date_start + datetime.timedelta(days=31),
```

Comment:
The end date of the first subscription should be set to the starting date of the next subscription.  Otherwise there will be a one-day gap.